powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java

```diff
@@ -1962,12 +1940,58 @@ private ActivationRecovery createRecoveryCodeForActivation(ActivationRecordEntit
         }
     }
 
+    /**
+     * Internal logic for processing activation removal.
+     * @param activation Activation entity.
+     * @param externalUserId External user identifier.
+     * @param revokeRecoveryCodes Whether associated recovery codes should be revoked.
+     */
+    private void removeActivationInternal(ActivationRecordEntity activation, String externalUserId, boolean revokeRecoveryCodes) {
+        activation.setActivationStatus(io.getlime.security.powerauth.app.server.database.model.ActivationStatus.REMOVED);
+        // Recovery codes are revoked in case revocation is requested, or always when the activation is in CREATED or PENDING_COMMIT state
+        revokeRecoveryCodes = revokeRecoveryCodes
+                || activation.getActivationStatus() == ActivationStatus.CREATED
+                || activation.getActivationStatus() == ActivationStatus.PENDING_COMMIT;
+        if (revokeRecoveryCodes) {
+            revokeRecoveryCodes(activation.getActivationId());
+        }
```

Comment:
Please, let's not assign values to the parameter.

```suggestion
    private void removeActivationInternal(final ActivationRecordEntity activation, final String externalUserId, final boolean revokeRecoveryCodes) {
        activation.setActivationStatus(io.getlime.security.powerauth.app.server.database.model.ActivationStatus.REMOVED);
        // Recovery codes are revoked in case revocation is requested, or always when the activation is in CREATED or PENDING_COMMIT state
        if (revokeRecoveryCodes
                || activation.getActivationStatus() == ActivationStatus.CREATED
                || activation.getActivationStatus() == ActivationStatus.PENDING_COMMIT) {
            revokeRecoveryCodes(activation.getActivationId());
        }
```